### ISSUE TITLE
Parameter autonormalization

### ISSUE THREAD

**Issue by SebsCubs** [JBjoernskov/Twin4Build#84]
[tps.Parameter.set()](https://github.com/JBjoernskov/Twin4Build/blob/61322cff77e70238146006e974885eca8d02a28d/twin4build/utils/types.py#L767C5-L774C42) enforces normalization before storage. This complicates loading `EstimationResults`, as [set_parameters_from_array()](https://github.com/JBjoernskov/Twin4Build/blob/61322cff77e70238146006e974885eca8d02a28d/twin4build/model/simulation_model/simulation_model.py#L2111C11-L2111C79) (called by [load_estimation_result()](https://github.com/JBjoernskov/Twin4Build/blob/61322cff77e70238146006e974885eca8d02a28d/twin4build/model/simulation_model/simulation_model.py#L2030C5-L2111C79)) relies on default min/max values for this normalization rather than the original boundaries.

I am not sure why the parameter value is stored always normalized and whether this is important for other functionality. But to manage parameter estimation results between model instances we would need to either:

1. Store unnormalized values (requires refactoring `Parameter`, my preference).

2. Modify the set_parameters_from_array() method to require and pass min/max boundaries. (would require tracking and testing the places where this method is used)

@JBjoernskov What are your thoughts on this?

**Comment by JBjoernskov** [comment]
Lets go with option 1. I can see that we already save bounds after estimation, `lb=self._lb` and `ub=self._ub` used for normalization. 
If `tps.Parameter` is initialized without bounds, e.g. `tps.Parameter(torch.tensor(5.2))`, the normalization has no effect (because of the way we set the bounds in `tps.Parameter.__new__()`. 

Therefore, the model should behave correctly if you use `set_parameters_from_array()` with unnormalized values - even if we dont have bounds. 

Something like:

```python

# **Estimation**

# Get unnormalized parameter values
result_x = []
for obj, attr in zip(self._flat_components, self._parameter_names):
    obj = rgetattr(self._components[obj], attr)
    v = obj.get()
    result_x.append(v)

# Create and save result
result = EstimationResult(
      result_x=result_x
      component_id=[com.id for com in self._flat_components],
      component_attr=[attr for attr in self._parameter_names],
      theta_mask=self._theta_mask,
      start_time=self._start_time,
      end_time=self._end_time,
      step_size=self._stepSize,
      x0=self._x0,
      lb=self._lb,
      ub=self._ub,
      iterations=getattr(result, "nit", None),
      nfev=getattr(result, "nfev", None),
      final_objective=getattr(result, "fun", None),
      success=getattr(result, "success", None),
      message=getattr(result, "message", None),
)
```

FYI, the seemingly complicated logic in `set_parameters_from_array()` that includes both `tps.Parameter` and `tps.TensorParameter` is to support both torch solvers (Adam, SGD, etc.) and scipy solvers for estimation. 

`tps.Parameter()` is used for torch solvers (inherits from torch.nn.Parameter).
`tps.TensorParameter`  is used for scipy solvers.

**Comment by SebsCubs** [comment]
@JBjoernskov I found the culprit of my confusion:

When the model components are instantiated (for example [thermal space](https://github.com/JBjoernskov/Twin4Build/blob/61322cff77e70238146006e974885eca8d02a28d/twin4build/systems/building_space/building_space_thermal_torch_system.py#L233C1-L249C7) ) they provide default values for parameters, by doing so they also define min and max values indirectly in the [__new__ method of Parameter](https://github.com/JBjoernskov/Twin4Build/blob/61322cff77e70238146006e974885eca8d02a28d/twin4build/utils/types.py#L643) And then when I am trying to load the parameters from a file, the boundaries are set from the default values.

I think the best way forward would be to then modify the `set_parameters_from_array()` method to also update the boundaries in the Parameter object.

**Comment by JBjoernskov** [comment]
I would be fine with introducing 2 more optional kwargs, i.e.: `set_parameters_from_array(..., lb=None, ub=None)`.
And set these if provided.
I agree that having default bounds in `Parameter` is confusing.  As these methods are currently public (another question is whether this should be the case), this behavior could lead to unexpected results. It would make sense to let the user update these bounds - even if there is no current use-cases (unless the user is doing some custom stuff and explicitly calls `Parameter.normalize()` and `Parameter.denormalize()`).

Currently, the only use-case of `Parameter.normalize()` and `Parameter.denormalize()` is in `Estimator`.
And whenever we call the `Estimator.estimate()`, the bounds are overwritten. 
This change is therefore not expected to change any behavior. 


Suggestion:
Maybe, we should make `min_value` and `max_value` required arguments in `Parameter.__new__`. Then in [thermal space](https://github.com/JBjoernskov/Twin4Build/blob/61322cff77e70238146006e974885eca8d02a28d/twin4build/systems/building_space/building_space_thermal_torch_system.py#L233C1-L249C7) for instance (and all other System classes), we would specify sensible default values. 
The user can then overwrite these when calling `Estimator.estimate()` or omit the bounds and use the default values. This would simplify the use of the API for unexperienced users.

**Comment by SebsCubs** [comment]
This issue was addressed in this commit https://github.com/JBjoernskov/Twin4Build/commit/22ad14a6394c2dbff0d45d443aa4cae84bf9faa4 together with a new test to fix this. 
As a general rule, the parameters should be loaded from a file before any simulation since the estimator returns a model with the original parameters, not the resulting ones.
